### PR TITLE
Share certificate links for writes_disabled visitors

### PR DESCRIPTION
**Pull request by bcjordan**
When `hoc_activity_writes_disabled` is set, the certificate page shows up differently.
# When identifier is passed ("non-anonymous")

https://code.org/congrats?i=aa2c3359fcef07b91d09b32d81d45240&s=bWM=

A url with the identifier (`aa2c3359fcef07b91d09b32d81d45240` in that example) is immediately rendered to the page, and the (name or generated certificate?) is later set to that identifier.

"I just did the #HourOfCode - check it out! @codeorg http://code.org/certificates/aa2c3359fcef07b91d09b32d81d45240"

This PR should not change the logic of this non-anonymous code path.
# When identifier is not passed ("anonymous")

e.g. http://code.org/congrats?s=bWM=

Previously, we would set the share links to reference  http://code.org/certificates/blank and never update when the user enters their name and their certificate link is generated.

"I just did the #HourOfCode - check it out! @codeorg http://code.org/certificates/blank"

This change swaps out new share links with updated certificate URLs.

Now:

"I just did the #HourOfCode - check it out! @codeorg http://brian-testmc.code.org/v2/hoc/certificate/eyJuYW1lIjoiSSBhbSBCcmlhbiIsImNvdXJzZSI6Im1jIn0=.jpg"

### CODE REVIEW

File: pegasus/sites.v3/code.org/views/congrats_share.haml [hunk 1]
```diff
@@ -20,12 +20,36 @@
...
+
+    return 'https://www.facebook.com/sharer/sharer.php?' + $.param(queryParams, true);
+  }
+
+  function generateTwitterURL(urlToShare) {
+    var queryParams = {
+      url: urlToShare,
+      related: 'codeorg',
+      hashtags: 'HourOfCode', // TODO(bjordan): anon certs may not always be HoC
+      text: "#{@just_did_hoc_text}"
```
Comment (bcjordan) [inline pegasus/sites.v3/code.org/views/congrats_share.haml]: I'm not super excited about passing the text using HAML interpolation into the JS function but it felt less risky than making assumptions about javascript variable scope / setting order.

File: pegasus/sites.v3/code.org/views/congrats_share.haml [hunk 2]
```diff
@@ -20,12 +20,36 @@
...
+    };
+
+    return 'https://www.facebook.com/sharer/sharer.php?' + $.param(queryParams, true);
+  }
+
+  function generateTwitterURL(urlToShare) {
+    var queryParams = {
+      url: urlToShare,
+      related: 'codeorg',
+      hashtags: 'HourOfCode', // TODO(bjordan): anon certs may not always be HoC
```
Comment (mehalshah) [inline pegasus/sites.v3/code.org/views/congrats_share.haml]: Do you still need to do this if just_did_hoc_text includes #HourOfCode?

Comment (bcjordan) [inline pegasus/sites.v3/code.org/views/congrats_share.haml]: Ah, great catch, removing that is the more sane default.

**Comment by bcjordan** [comment]
Going to implement conditional hashtag inclusion for i18n strings

**Comment by bcjordan** [comment]
Closing in favor of https://github.com/code-dot-org/code-dot-org/pull/5797 approach

**Pull request closed without merging**